### ISSUE TITLE
[BUG] stack overflow with zero-width stroke

### ISSUE THREAD

**Issue by CinusMinus** [38/plotters#267]
**Describe the bug**
Attempting to draw a zero-width line will cause a stack overflow with plotters 0.3.1. Expected behaviour: Nothing is drawn. Maybe something very thin. But no stack overflow.

**To Reproduce**
`main.src`
```rust
use plotters::prelude::*;

fn main() {
    let p1 = (338, 122);
    let p2 = (365, 122);

    let mut backend = BitMapBackend::new("blub.png", (800, 600));

    backend.draw_line(p1, p2, &RGBColor(0, 0, 0).stroke_width(0)).unwrap();
}
```
`Cargo.toml`
```toml
[package]
name = "plottest"
version = "0.1.0"
edition = "2018"

[dependencies]
plotters = "0.3.1"
```
Result:
```
> cargo run
thread 'main' has overflowed its stack
fatal runtime error: stack overflow
[1]    123 abort (core dumped)  cargo run
```
If I change the line width to 1 it works.

**Version Information**
plotters 0.3.1

**Comment by 38** [comment]
Hi there, thanks for reporting this. The issue should be solve by [fixing the plotters-backend crate](https://github.com/plotters-rs/plotters-backend/commit/7351b8ef6d066ad5e3368100aea712a1d4c758c1). The fix is already published in plotters-backend 0.3.1. You should be able to get the problem solved by running `cargo update`.

Please let me know if the fix works.

Thanks!

**Comment by CinusMinus** [comment]
Yes, works now. Thanks!
